### PR TITLE
chore(cli): migrate error component to typescript

### PR DESCRIPTION
**Pull request by Naturalclar**
## Description

Migrated `Error` component in the cli to typescript

## Related Issues

#21995

### CODE REVIEW

File: packages/gatsby-cli/src/reporter/loggers/ink/components/error.tsx [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+  )
+}
+
+interface IDocsLinkProps {
+  docsUrl: string
+}
+const DocsLink = ({ docsUrl }: IDocsLinkProps): JSX.Element => {
+  // TODO: when there's no specific docsUrl, add helpful message describing how
+  // to submit an issue
+  if (docsUrl === `https://gatsby.dev/issue-how-to`) return null
```
Comment (blainekasten) [inline packages/gatsby-cli/src/reporter/loggers/ink/components/error.tsx]: I think the return type of DocsLink needs to be `React.ReactNode` in order to support null

File: packages/gatsby-cli/src/reporter/loggers/ink/components/error.tsx [hunk 2]
```diff
@@ -0,0 +1,99 @@
+import React from "react"
+import path from "path"
+import { Color, Box } from "ink"
+import { get } from "lodash"
+
+interface IFileProps {
+  filePath: string
+  location: string
+}
+const File = ({ filePath, location }: IFileProps): JSX.Element => {
```
Comment (MichaelDeBoey) [inline packages/gatsby-cli/src/reporter/loggers/ink/components/error.tsx]:
```suggestion
const File: FunctionComponent<IIFileProps> = ({ filePath, location }) => {
```

File: packages/gatsby-cli/src/reporter/loggers/ink/components/error.tsx [hunk 3]
```diff
@@ -0,0 +1,99 @@
+import React from "react"
```
Comment (MichaelDeBoey) [inline packages/gatsby-cli/src/reporter/loggers/ink/components/error.tsx]:
```suggestion
import React, { FunctionComponent, memo } from "react"
```

File: packages/gatsby-cli/src/reporter/loggers/ink/components/error.tsx [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+      {path.relative(process.cwd(), filePath)}
+      {locString}
+    </Color>
+  )
+}
+
+interface IDocsLinkProps {
+  docsUrl: string
+}
+const DocsLink = ({ docsUrl }: IDocsLinkProps): JSX.Element => {
```
Comment (MichaelDeBoey) [inline packages/gatsby-cli/src/reporter/loggers/ink/components/error.tsx]:
```suggestion
const DocsLink: FunctionComponent<IDocsLinkProps> = ({ docsUrl }) => {
```

File: packages/gatsby-cli/src/reporter/loggers/ink/components/error.tsx [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+    type?: string
+    text: string
+    filePath?: string
+    location: string
+    docsUrl: string
+  }
+}
+
+export const Error = React.memo(
+  ({ details }: IErrorProps): JSX.Element => (
```
Comment (MichaelDeBoey) [inline packages/gatsby-cli/src/reporter/loggers/ink/components/error.tsx]:
```suggestion
export const Error = memo<IErrorProps>(({ details }) => (
```

File: packages/gatsby-cli/src/reporter/loggers/ink/components/error.tsx (hunk 2 above)
Comment (blainekasten) [reply packages/gatsby-cli/src/reporter/loggers/ink/components/error.tsx]: I like the other approach better

Comment (cola119) [reply packages/gatsby-cli/src/reporter/loggers/ink/components/error.tsx]: I want to unify which statement should be used, `React.FC` or `React.FunctionComponent`.
https://github.com/gatsbyjs/gatsby/pull/22084#discussion_r390042955

Comment (MichaelDeBoey) [reply packages/gatsby-cli/src/reporter/loggers/ink/components/error.tsx]: @blainekasten Using `FunctionComponent` will give you correcter types (it includes `children` for you) and will make sure you can't invoke `File` as a normal function, but can only use it as a component like `<File />`.

@cola@ I always use `FunctionComponent` so it's a lot clearer what it means

Comment (ascorbic) [reply packages/gatsby-cli/src/reporter/loggers/ink/components/error.tsx]: I have to agree with @MichaelDeBoey . `FunctionComponent` is the right way to do this.

**Comment by Naturalclar** [comment]
Thanks for the review! Changed the Component Type to FunctionComponent

**Pull request merged**